CMSIS/NN/Source/SoftmaxFunctions/arm_softmax_s8.c

```diff
@@ -48,12 +89,118 @@ void arm_softmax_s8(const int8_t *input,
                     const int8_t diff_min,
                     int8_t *output)
 {
+#ifdef ARM_MATH_MVEI
+
+#define ACT_MIN (-128)
+#define ACT_MAX (127)
+
+    const int32_t mask = (1 << shift);
+
+    for (int i_num_rows = 0; i_num_rows < num_rows; ++i_num_rows)
+    {
+        int8_t max = ACT_MIN;
+
+        int32_t vec_count = row_size / 16;
```

Comment:
Since the division could be expensive, what about:
        for (int i = 0; i <= (vec_size - 16); i+=16)
        {
            const int8x16_t ip = vldrbq_s8(&input[i]);
            max = vmaxvq_s8(max, ip);
        }

In this manner you can also avoid the multiplication by 16 for calculating the array index

Reply:
@gmiodice For the division, I don't think so as it is  a divide by a power of 2 number which should be resolved as  a shift operation. Normally, I would have done it as row_size >> 4, but for clarity i have gone with /16 and trust the compiler to do the rest. With that in mind, do want to change to avoid the multiplication alone?

Reply:
So it is ok to keep it as it is. ;)